quipucords/api/report/renderer.py

```diff
@@ -138,19 +147,51 @@ def render(self,
         if not systems_list:
             return None
         csv_writer.writerow(['Report:'])
-
         headers = csv_helper.generate_headers(
-            systems_list, exclude=set([
-                'id', 'report_id', 'metadata', 'entitlements']))
+            systems_list,
+            exclude={'id', 'report_id', 'metadata', 'entitlements'})
+        if self.SOURCES_KEY in headers:
+            headers += self.source_headers
+            headers = sorted(list(set(headers)))
+
+        # Add source heaaders
         csv_writer.writerow(headers)
         for system in systems_list:
             row = []
+            system_sources = system.get(self.SOURCES_KEY)
+            if system_sources is not None:
+                sources_info = self._compute_source_info(system_sources)
+            else:
+                sources_info = None
             for header in headers:
-                fact_value = system.get(header)
+                fact_value = None
+                if header in self.source_headers:
+                    if sources_info is not None:
+                        fact_value = sources_info.get(header)
+                else:
+                    fact_value = system.get(header)
                 row.append(csv_helper.serialize_value(header, fact_value))
             csv_writer.writerow(row)
 
         csv_writer.writerow([])
 
         csv_content = report_buffer.getvalue()
         return csv_content
+
+    def _compute_source_info(self, sources):
+        """Detect scan source types."""
+        result = {
+            self.NETWORK_DETECTION_KEY: False,
```

Comment:
is `self.` necessary?

Reply:
sure

Reply:
Yeah, we probably have the same constants in many files.  It may be worth making a constants file at some time.